sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -196,6 +203,27 @@ export class EventHubReceiver extends LinkEntity {
       sequenceNumber: -1
     };
     this._onAmqpMessage = (context: EventContext) => {
+      this._receivedMessageCount++;
+      const logMessageInterval = 1000;
+      // log every 1000 messages
+      if (this._receivedMessageCount % logMessageInterval === 1) {
+        log.receiver(
+          "[%s] Receiver '%s' has received %i messages.",
+          this._context.connectionId,
+          this.name,
+          this._receivedMessageCount
+        );
+      }
+      if (this._receivedMessageCount >= Number.MAX_SAFE_INTEGER) {
+        log.receiver(
+          "[%s] Receiver '%s' has processed %i messages. Resetting message count.",
+          this._context.connectionId,
+          this.name,
+          this._receivedMessageCount
+        );
+        this._receivedMessageCount = 0;
```

Comment:
Why not reset every 1000 events?
We dont need to know the number of events received, we only need a heartbeat of sorts.
And an indication of probably the last time this was logged?

I would suggest something like `received 1000 messages since <last-timestamp-here>`?

Reply:
That sounds like a more reasonable approach.